core/options.js

```diff
@@ -37,10 +38,9 @@ Blockly.Options = function(options) {
     var hasDisable = false;
     var hasSounds = false;
   } else {
-    var languageTree =
-        Blockly.Options.parseToolboxTree(options['toolbox'] || null);
-    var hasCategories = Boolean(languageTree &&
-        languageTree.getElementsByTagName('category').length);
+    var languageTree = Blockly.Options.parseToolboxTree(options['toolbox'] || null);
+    languageTree = Blockly.utils.toolbox.parseToolboxContents(languageTree);
```

Comment:
I have a similar concern here about reusing a variable: if the first and second lines are doing different things, and return different types, you should declare a new variable instead of reusing the old one.